src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTests.vb

```diff
@@ -7819,8 +7868,65 @@ DERIVED: Public Overridable Sub f(x As Integer, Optional y As String = "")
 ]]>)
         End Sub
 
+        <Fact()>
+        Public Sub OverridingFunctionsOverloadedOnOptionalParameters_B()
+            Dim useOpts = _ImplicitDefaultOptionalParameter_
+            If Not Syntax.InternalSyntax.Parser.CheckFeatures(Syntax.InternalSyntax.Feature.ImplicitDefaultValueOnOptionalParameter, useOpts) Then
+                Assert.True(False, $"Feature{NameOf(Syntax.InternalSyntax.Feature.ImplicitDefaultValueOnOptionalParameter)} is not present")
+            End If
+            CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+Imports System
+Class base
+    Public Overridable Sub f(x As Integer)
+        Console.WriteLine("BASE: Public Overridable Sub f(x As Integer)")
+    End Sub
+    Public Overridable Sub f(x As Integer, Optional y As Integer)
+        Console.WriteLine("BASE: Public Overridable Sub f(x As Integer, Optional y As Integer)")
+    End Sub
+    Public Overridable Sub f(x As Integer, Optional y As String = "")
+        Console.WriteLine("BASE: Public Overridable Sub f(x As Integer, Optional y As String = """")")
+    End Sub
+End Class
+
+Class derived
+    Inherits base
+    Public Overrides Sub f(x As Integer)
+        Console.WriteLine("DERIVED: Public Overridable Sub f(x As Integer)")
+    End Sub
+    Public Overrides Sub f(x As Integer, Optional y As Integer)
```

Comment:
For example
Scenario 1:
base - `Public Overridable Sub f(x As Integer, Optional y As Integer)```
derived -```Public Overrides Sub f(x As Integer, Optional y As Integer = 0)```

Scenario 2:
base - `Public Overridable Sub f(x As Integer, Optional y As Integer = 0)```
derived -```Public Overrides Sub f(x As Integer, Optional y As Integer)```

Scenario 3:
base - `Public Overridable Sub f(x As Integer, Optional y As Integer)```
derived -```Public Overrides Sub f(x As Integer, Optional y As Integer = 1)```

Scenario 4:
base - `Public Overridable Sub f(x As Integer, Optional y As Integer = 2)```
derived -```Public Overrides Sub f(x As Integer, Optional y As Integer)```